src/components/App.tsx

```diff
@@ -1798,6 +1798,17 @@ class App extends React.Component<AppProps, AppState> {
         });
       }
 
+      // remove selections on escape press
+      if (event.key === "Escape") {
```

Comment:
Other than the suggested changes, this looks good!
```suggestion
      if (event.key === KEYS.ESCAPE) {
        // Don't deselect text right after editing it
        if (isWritableElement(event.target)) {
          return;
        }
```